### PR TITLE
feat: allow setting ChannelFlags directly when editing a channel or thread

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary
```py
await channel.edit(flags=disnake.ChannelFlags(pinned=True)
```
## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/channel.py [hunk 1]
```diff
@@ -365,6 +365,7 @@ async def edit(
         default_auto_archive_duration: Optional[AnyThreadArchiveDuration] = MISSING,
         type: ChannelType = MISSING,
         overwrites: Mapping[Union[Role, Member], PermissionOverwrite] = MISSING,
+        flags: ChannelFlags = MISSING,
```
Comment (shiftinv) [inline disnake/channel.py:368]: missing in overload

File: disnake/channel.py [hunk 2]
```diff
@@ -416,6 +417,11 @@ async def edit(
         default_auto_archive_duration: Optional[Union[:class:`int`, :class:`ThreadArchiveDuration`]]
             The new default auto archive duration in minutes for threads created in this channel.
             Must be one of ``60``, ``1440``, ``4320``, or ``10080``.
+        flags: :class:`ChannelFlags`
+            The new flags to set for this channel.
```
Comment (shiftinv) [inline disnake/channel.py:421]: This should clarify that all old flags will be overwritten - the current description could be interpreted as setting new flags while keeping existing ones.

File: disnake/channel.py [hunk 3]
```diff
@@ -2155,6 +2178,7 @@ async def edit(
         position: int = MISSING,
         nsfw: bool = MISSING,
         overwrites: Mapping[Union[Role, Member], PermissionOverwrite] = MISSING,
+        flags: ChannelFlags = MISSING,
```
Comment (shiftinv) [inline disnake/channel.py:2181]: missing in overload

**Review by shiftinv — COMMENTED** [review]
Looks good!
Is setting flags also supported on channel creation, or just when editing?

**Comment by onerandomusername** [comment]
Going to add support for setting flags on creation as well.

**Comment by onerandomusername** [comment]
> Going to add support for setting flags on creation as well.

Turns out the API doesn't support this, oops.

File: disnake/threads.py [hunk 4]
```diff
@@ -685,10 +691,15 @@ async def edit(
...
         if slowmode_delay is not MISSING:
             payload["rate_limit_per_user"] = slowmode_delay
+
         if pinned is not MISSING:
-            flags = ChannelFlags._from_value(self._flags)
+            # create base flags if flags are provided, otherwise use the internal flags.
+            if flags is None:
+                flags = ChannelFlags._from_value(self._flags)
             flags.pinned = pinned
             payload["flags"] = flags.value
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/threads.py]: https://github.com/DisnakeDev/disnake/pull/642/commits/615e105c3c93a8235c291b158991b1beb124196b

File: disnake/channel.py (hunk 1 above)
Comment (onerandomusername) [reply disnake/channel.py:369]: https://github.com/DisnakeDev/disnake/pull/642/commits/615e105c3c93a8235c291b158991b1beb124196b

File: disnake/threads.py [hunk 5]
```diff
@@ -685,10 +691,15 @@ async def edit(
             payload["invitable"] = invitable
         if slowmode_delay is not MISSING:
             payload["rate_limit_per_user"] = slowmode_delay
+
         if pinned is not MISSING:
-            flags = ChannelFlags._from_value(self._flags)
+            # create base flags if flags are provided, otherwise use the internal flags.
+            if flags is None:
+                flags = ChannelFlags._from_value(self._flags)
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/threads.py]: https://github.com/DisnakeDev/disnake/pull/642/commits/615e105c3c93a8235c291b158991b1beb124196b

File: disnake/channel.py (hunk 3 above)
Comment (onerandomusername) [reply disnake/channel.py:2183]: https://github.com/DisnakeDev/disnake/pull/642/commits/615e105c3c93a8235c291b158991b1beb124196b

File: disnake/threads.py [hunk 6]
```diff
@@ -620,6 +620,7 @@ async def edit(
         slowmode_delay: int = MISSING,
         auto_archive_duration: AnyThreadArchiveDuration = MISSING,
         pinned: bool = MISSING,
+        flags: ChannelFlags = None,
```
Comment (shiftinv) [inline disnake/threads.py:623]:
```suggestion
        flags: ChannelFlags = MISSING,
```
edit methods should generally use `MISSING` for all parameter defaults (except `reason`) - this also needs some slight adjustments in the parameter handling below

Comment (onerandomusername) [reply disnake/threads.py]: https://github.com/DisnakeDev/disnake/pull/642/commits/e62f2da4ecaa0aed7798ed81e5d803f4323074fe

**Review by shiftinv — APPROVED** [review]
lgtm!

**Pull request merged**